client/src/pages/Dashboard/Dashboard.tsx

```diff
@@ -10,8 +10,12 @@ export default function Dashboard(): JSX.Element {
   const classes = useStyles();
   const { initSocket } = useSocket();
 
+
   useEffect(() => {
```

Comment:
In regards to initializing the socket, you can do this in the sockets context within a useEffect (check to see if user is logged in and only then initialize the socket, you can also add the loggedInUser as a dep to the useEffect).